### ISSUE TITLE
Bad Request when using divideSeries 

### ISSUE THREAD

**Issue by cjagus** [go-graphite/carbonapi#388]
running carbonapi 0.11.0, getting below error when using divideSeries function, same working fine in graphite-web 

`
2019-04-17T13:30:48.305Z	ERROR	access	request failed	{"data": {"handler":"render","carbonapi_uuid":"d7af41e6-3e89-4818-8004-00fa76ffe674","url":"/render","peer_ip":"127.0.0.1","host":"carbonapi","format":"json","use_cache":true,"targets":["alias(divideSeries(sumSeries(metrics.apps-recs-supervisor.*.counters.realtime-recs.rts.get_parameters.thrift_timeout.*.count), sumSeries(metrics.apps-recs-supervisor.*.counters.realtime-recs.rts.get_parameters.call.*.count)), 'Thrift timeouts')","alias(divideSeries(sumSeries(metrics.apps-recs-supervisor.*.counters.realtime-recs.rts.get_parameters.tornao_timeout.*.count), sumSeries(metrics.apps-recs-supervisor.*.counters.realtime-recs.rts.get_param.call.*.count)), 'Tornado Timeouts')","alias(divideSeries(#E, #B), 'TTransportException')","alias(divideSeries(sumSeries(metrics.apps-recs-supervisor.*.counters.realtime-recs.rts.get_param.fallback.*.count), sumSeries(metrics.apps-recs-supervisor.*.counters.real-recs.rts.get_parameters.call.*.count)), 'Fallback parameters config response')","alias(divideSeries(#J, sumSeries(metrics.apps-recs-supervisor.*.counters.real-recs.rts.get_param.call.*.count)), 'Timeouts')"],"cache_timeout":60,"metrics":["metrics.apps-recs-supervisor.*.counters.real-recs.rts.get_param.thrift_timeout.*.count","metrics.apps-recs-supervisor.*.counters.real-recs.rts.get_param.call.*.count","metrics.apps-recs-supervisor.*.counters.realtime-recs.rts.get_param.tornao_timeout.*.count","metrics.apps-recs-supervisor.*.counters.real-recs.rts.get_param.call.*.count"],"runtime":3.920347289,"http_code":400,"reason":"Bad Request\n\nTarget              : alias(divideSeries(#E, #B), 'TTransportException')\nError               : missing argument\nParsed so far       : alias(divideSeries(\nCould not parse     : #E, #B), 'TTransportException')\n","from":1555504244,"until":1555507844,"from_raw":"-1h","until_raw":"now","uri":"/render","zipper_requests":1987}}
`
@Civil

**Comment by Civil** [comment]
Ok, apparently I need to make 'allowed symbols' customizable and make the default compatible with Graphite. Or just to ignore "not allowed symbols" for everything under 'alias' function.

**Comment by Civil** [comment]
For a temporary workaround you can manually add "#" here: https://github.com/go-graphite/carbonapi/blob/c152600d64e6f9e137fe21d7ab63ce16326b31cf/pkg/parser/parser.go#L477

**Comment by cjagus** [comment]
adding "#"  char fixed the issue, is it recommended to run current master in production or should I build from 0.11.0 tag @Civil

**Comment by cjagus** [comment]
any thoughts on this @Civil ^

**Comment by Civil** [comment]
Well... I'm no longer working at Booking.com, so I no longer have access to their graphite setup. So all the work I do I can only test at home.

We also had an argue with the people who are now maintaining graphite there and we took fundamentally different paths starting from 0.11.

They have their own fork bookingcom/carbonapi - which they use in their production environment and they've focused on features they've need (monitoring it with Prometheus for example).

I've focused on other stuff like merging carbonapi with zipper to make it easier to use, to allow other backends, not only carbonapi-compatible (current master and 0.12.0-rc.0 have support for msgpack as backend and for Prometheus as a backend, it also supports different balancing modes that's useful for storages that can do their own clustering like grafana/metrictank or lomik/graphite-clickhouse). This version also have tags support for example and can proxy functions that it doesn't support to graphite-web.

However as my setup is relatively small and I do all the work on a project during my free time, I cannot guarantee anything about it. I would recommend to test it and decide for yourself.
